Source/JavaScriptCore/runtime/ScriptExecutable.cpp

```diff
@@ -268,6 +273,17 @@ CodeBlock* ScriptExecutable::newCodeBlockFor(CodeSpecializationKind kind, JSFunc
         RELEASE_ASSERT(kind == CodeForCall);
         RELEASE_ASSERT(!executable->m_codeBlock);
         RELEASE_ASSERT(!function);
+
+        UnlinkedModuleProgramCodeBlock* unlinkedCodeBlock = executable->unlinkedCodeBlock();
+        ParserError error;
+        if (UNLIKELY(!unlinkedCodeBlock))
+            unlinkedCodeBlock = executable->generateOrGetUnlinkedCodeBlock(globalObject, throwScope, error);
+
+        if (UNLIKELY(!unlinkedCodeBlock)) {
+            throwException(globalObject, throwScope, error.toErrorObject(globalObject, executable->source()));
+            return nullptr;
+        }
```

Comment:
You can just `RELEASE_ASSERT(unlinkedCodeBlock)` instead in the if block after `generateOrGetUnlinkedCodeBlock` above.